src/main/scala/ru/otus/sc/storage/service/impl/StorageServiceImpl.scala

```diff
@@ -0,0 +1,15 @@
+package ru.otus.sc.storage.service.impl
+
+import ru.otus.sc.storage.dao.StorageDao
+import ru.otus.sc.storage.model.{StorageRequest, StorageResponse}
+import ru.otus.sc.storage.service.StorageService
+
+class StorageServiceImpl(dao: StorageDao) extends StorageService {
+
+  def get(request: StorageRequest): Option[StorageResponse] = {
+    val value = dao.get(request.storageKey)
+    // Can not use pattern matching or .map, so use if/else
+    if (value.nonEmpty) Some(StorageResponse(value.get))
+    else None
```

Comment:
```suggestion
    value match {
      case Some(value) => Some(StorageResponse(value))
      case None => None
    }
```

`Option.get` нужно всегда избегать. Это принцип "локального мышления" ([local reasoning](https://degoes.net/articles/fp-glossary))

Но ещё лучше здесь было бы использовать функциональный комбинатор:

```suggestion
    value.map(x => StorageResponse(x))
```

Reply:
Абсолютно верно, но в ДЗ1 мы "понятия не имеем" ни о патерн матчинге, ни о .map, это даже указано в коментах специально, поэтому это так